File: ABS/ABC081B.py
n = int(input())
n_list = list(map(int, input().split()))

count = 0
flg = True

while flg:
    for i in range(n):
        if n_list[i] % 2 == 1:
            flg = False
            print(count)
            break
        n_list[i] = n_list[i] // 2
    else:
        count += 1

File: ABS/ABC081B.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a.at(i);
    }
    int count = 0;
    bool flg = true;

    while (flg) {
        for (int i = 0; i < n; i++) {
            if (a.at(i) % 2 == 0) {
                a.at(i) /= 2;
            } else {
                flg = false;
            }
        }
        if (flg) {
            count ++;
        }
    }

    cout << count << endl;
    return 0;
}
